tests/neg/OrType.scala

```diff
@@ -2,7 +2,7 @@ class B(val x: Int)
 class C(val x: Double)
 
 object Test{
-  def bar(x: B | C): Int | Double = x.x
+  def bar(x: B | C): Int | Double = x.x  // error
```

Comment:
@smarter Both examples make use of the fact that the alternatives have "coincidentally" members with the same type. Both also require possibly expensive splitting under the cover. Maybe it's more honest to require one match for `Assign` and another for `Plus`. For now, it looks to me like the use cases are close to code smells.